### PR TITLE
Document temporal requirements

### PR DESCRIPTION
**Pull request by shumbo**
This PR adds documentation for temporal requirements.

Things to discuss:

- How do we want to update the dynamic scenarios tutorial?

### CODE REVIEW

File: docs/reference/dynamic_scenarios.rst [hunk 1]
```diff
@@ -65,7 +65,7 @@ In detail, a single time step of a dynamic simulation is executed according to t
 
 9. Update every :term:`dynamic property` of every object to its current value in the simulator.
 
-10. If the simulation is stopping for one of the reasons above, first check if any of the :keyword:`require eventually` conditions of any remaining scenarios were never satisfied: if so, reject the simulation.
+10. If the simulation is stopping for one of the reasons above, first check if any of the liveness properties of any remaining scenarios were never satisfied: if so, reject the simulation.
```
Comment (shumbo) [inline docs/reference/dynamic_scenarios.rst:68]: I didn't understand what this meant but systematically replaced "require eventually conditions" with "liveness properties". I want to make sure this makes sense.

File: docs/reference/operators.rst [hunk 2]
```diff
@@ -154,3 +154,51 @@ The midpoint of the corresponding edge of the Object’s bounding box, inheritin
...
+.. _next:
+
+next *condition*
+----------------
+Require the given condition to hold at the next time step of the dynamic scenario.
+
+.. _{condition} until {condition}:
+.. _until:
+
+*condition* until *condition*
```
Comment (shumbo) [inline docs/reference/operators.rst:190]: Is there a name for LHS and RHS of the until operator, like "hypothesis" and "conclusion" for `implies`?

File: docs/reference/dynamic_scenarios.rst (hunk 1 above)
Comment (Eric-Vin) [reply docs/reference/dynamic_scenarios.rst:68]: Makes sense to me! A liveness property is something that must occur in an execution, so "require eventually" is how we implement such a property.

File: docs/reference/operators.rst (hunk 2 above)
Comment (Eric-Vin) [reply docs/reference/operators.rst:190]: The closest I can think of would be precondition/postcondition, but this doesn't seem to exactly meet their definitions in the contexts I've heard them used.

Comment (dfremont) [reply docs/reference/operators.rst:190]: I don't know of specific names for these -- what you have is clear, I think.

File: docs/reference/operators.rst [hunk 3]
```diff
@@ -154,3 +154,51 @@ The midpoint of the corresponding edge of the Object’s bounding box, inheritin
...
+.. _{condition} implies {condition}:
+.. _implies:
+
+*hypothesis* implies *conclusion*
+---------------------------------
+Require the hypothesis to hold whenever the conclusion holds. This is a syntax sugar for :scenic:`not {hypothesis} or {conclusion}`.
+
+.. rubric:: References
+
+.. [B10] Bauer et al., :t:`Comparing LTL Semantics for Runtime Verification`, Journal of Logic and Computation. `[Online] <https://doi.org/10.1093/logcom/exn075>`_
```
[the remark this answers is not in the document]
Comment (dfremont) [reply docs/reference/operators.rst:204]: This is great. In fact I didn't know about the [title-reference role](https://docutils.sourceforge.io/docs/ref/rst/roles.html#title-reference), and will need to stick `:title:` in various other places in the docs!

File: docs/reference/operators.rst [hunk 4]
```diff
@@ -154,3 +154,51 @@ The midpoint of the corresponding edge of the Object’s bounding box, inheritin
...
+-----------------------------
+Require the first condition to hold until the second condition holds, at which point the first condition is no longer required to hold.
+This operator implements the strong semantics and requires the second condition to hold at some point during the execution.
+
+.. _{condition} implies {condition}:
+.. _implies:
+
+*hypothesis* implies *conclusion*
+---------------------------------
+Require the hypothesis to hold whenever the conclusion holds. This is a syntax sugar for :scenic:`not {hypothesis} or {conclusion}`.
```
Comment (dfremont) [inline docs/reference/operators.rst:200]: Should be "require the conclusion to hold whenever the hypothesis holds".

File: docs/reference/dynamic_scenarios.rst (hunk 1 above)
Comment (dfremont) [reply docs/reference/dynamic_scenarios.rst:68]: We should be careful here: a property like `(always X) and (eventually Y)` is neither a safety nor a liveness property (violations of it don't necessarily have a finite prefix after which the property can't be satisfied, nor is it always possible to extend a finite trace to one where the property is satisfied), but it needs to be checked at the end of the simulation. I'll revise this text.

File: docs/reference/operators.rst (hunk 4 above)
Comment (dfremont) [reply docs/reference/operators.rst:200]: Actually that sounds more like `always X implies Y`, so I'll rephrase.

**Comment by dfremont** [comment]
I've pushed some revisions, including updating the dynamics tutorial. If it looks OK to you we can merge.

**Pull request merged**